qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTests.java

```diff
@@ -121,20 +121,15 @@ public void test32JavaHomeOverride() throws Exception {
     public void test33RunsIfJavaNotOnPath() throws Exception {
         assumeThat(distribution().hasJdk, is(true));
 
-        final Result readlink = sh.run("readlink /usr/bin/java");
-        boolean unlinked = false;
-        try {
-            sh.run("unlink /usr/bin/java");
-            unlinked = true;
-
-            startElasticsearch(sh);
-            runElasticsearchTests();
-            stopElasticsearch(sh);
-        } finally {
-            if (unlinked) {
-                sh.run("ln -sf " + readlink.stdout.trim() + " /usr/bin/java");
-            }
+        // we don't require java be installed for the tests, but remove it if it's there
+        // since we don't require it for the tests, don't bother restoring it
+        if (Files.exists(Paths.get("/usr/bin/java"))) {
```

Comment:
Why was the previous code causing problems? If we are going to ensure /usr/bin/java doesn't exist, we should do so outside of test execution. This seems like an environmental issue, not something a test (in the middle of other tests running) should be changing. Deleting and not restoring system files is a serious, unexpected side effect to a test running.

Reply:
Vagrant VMs have a java package installed whereas CI workers do not. 
Since we label these as destructive and always run them on ephemeral VMs,  the restoring of this made the test unnecessarily hard to read.  I think it's worth restoring when it makes it possible to re-use a VM e.x. for development, but in this case, the test works if there's no java installed as well. 

The right fix here is probably to make sure java is not installed on the vagrant images ?

Reply:
I added the restore back to make this less awkward.